fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java

```diff
@@ -980,6 +981,30 @@ public void testIngestWithBinary() throws IOException {
         }
     }
 
+    /* Verifies RDF persisted as binary is retrieved with byte-for-byte fidelity */
+    @Test
+    public void testIngestOpaqueRdfAsBinary() throws IOException {
+        final HttpPost method = postObjMethod();
+        method.addHeader("Content-Type", "application/n-triples");
+        method.addHeader("Content-Disposition", "attachment");
+
+        final String rdf = "<test:/subject> <test:/predicate> <test:/object> .";
+        method.setEntity(new StringEntity(rdf));
+
+        try (final CloseableHttpResponse response = execute(method)) {
+            assertEquals(CREATED.getStatusCode(), getStatus(response));
+
+            final String location = getLocation(response);
+            final HttpGet get = new HttpGet(location);
+
+            try (final CloseableHttpResponse getResponse = execute(get)) {
+                final String resp = IOUtils.toString(getResponse.getEntity().getContent());
```

Comment:
Prefer `EntityUtils.toString` which is more specific and produces fewer short-lived objects. Although at this point I'm being really picky.